### PR TITLE
Add randgen tests

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <me@breeswish.org>

### CODE REVIEW

File: push-down-test/src/main.go [hunk 1]
```diff
@@ -96,15 +108,11 @@ func runTestCase(testCasePath string) {
 	go runStatements(pushDownLogChan, *connStrPush, statements)
 	go runStatements(pushDownWithBatchLogChan, *connStrPushWithBatch, statements)
 	ok := diffRunResult(testCasePath, noPushDownLogChan, pushDownLogChan, pushDownWithBatchLogChan)
-	if !ok {
-		log.Printf("Test failed. Stop.")
-		os.Exit(2) // Diff fail results in exit code 2 to distinguish with panic
-	}
+	return ok
```
Comment (lonng) [inline push-down-test/src/main.go]: Prefer `return diffRunResult(testCasePath, noPushDownLogChan, pushDownLogChan, pushDownWithBatchLogChan)`

File: push-down-test/src/main.go [hunk 2]
```diff
@@ -243,15 +272,17 @@ func diffRunResult(
 }
 
 func buildDefaultConnStr(port int) string {
-	return fmt.Sprintf("root@tcp(localhost:%d)/?allowNativePasswords=true", port)
+	return fmt.Sprintf("root@tcp(localhost:%d)/{db}?allowNativePasswords=true", port)
```
Comment (lonng) [inline push-down-test/src/main.go]: Why make this change.

Comment (lonng) [reply push-down-test/src/main.go]: Oh, got it. but I think replace is not a good idea. Why not use a place holder?

Comment (breeswish) [reply push-down-test/src/main.go]: It is a placeholder (by using replace to fill the place after that) 😅

Comment (breeswish) [reply push-down-test/src/main.go]: Do you have some suggestion to do it more elegantly?

**Pull request merged**